src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java

```diff
@@ -1,34 +1,50 @@
 package edu.kis.powp.jobs2d.drivers;
 
 import edu.kis.powp.jobs2d.Job2dDriver;
-import edu.kis.powp.jobs2d.LoggerDriver;
+import edu.kis.powp.jobs2d.drivers.extensions.ExtensionDriverComposite;
+import edu.kis.powp.jobs2d.drivers.extensions.UsageMonitorExtension;
 import edu.kis.powp.observer.Publisher;
 
 /**
  * Driver manager provides means to setup the driver. It also enables other
  * components and features of the application to react on configuration changes.
  */
 public class DriverManager {
-
-    private Job2dDriver currentDriver = new LoggerDriver();
+    private Job2dDriver currentDriver;
+    private ExtensionDriverComposite extensionDriver = new ExtensionDriverComposite();
     private Publisher changePublisher = new Publisher();
 
     /**
      * @param driver Set the driver as current.
      */
     public synchronized void setCurrentDriver(Job2dDriver driver) {
+        Job2dDriver temp = currentDriver;
         currentDriver = driver;
-        changePublisher.notifyObservers();
+        extensionDriver.setDriver(currentDriver);
+        if(temp instanceof UsageMonitorExtension) {
+            changePublisher.notifyObservers();
+        }
```

Comment:
observers should work without such hacks. If really necessary at least you should introduce "instanceof" on some general interface.